tasks/data.js

```diff
@@ -29,9 +29,38 @@ async function getCounts(context) {
 	}
 }
 
+// This is definitely a hack to achieve first adding items of collections that have dependencies in other collections i.e m2m, o2m
+// FIXME: Implement a more robust solution to sort collections based on their dependencies, or swap to a different way to seed the data
```

Comment:
I think we should look into migrating all the data first, and then "finishing" the schema by applying not-null and relationships. That should prevent any foreign key constraint / null constraint issues during the initial migration 🤔

Reply:
I actually tried doing this and had the migrating relations step last. 

There is two problems (apparently) with doing so: 

1. The Migrating Data step actually needs to know about the relations 
 - This is actually solvable, as we could easily grab the data about the relations before sending them to the v9 instance
 
2. The data migration fails due to it trying to insert data into a non existent column. (at least my dataset did this) 

More specifically: 

We have a Collection A, which has a m2m relation with Collection B, in a field called CollectionB. 

When I'm now running the data migration step, without having added the relations first, I get an error (from postgres, but probably doesn't matter about the db) that collectionA.collectionB is not a valid column (that can be added data to). I tried to understand what happens here by looking at directus source, but seemingly adding the relations also adjusts how directus (or the db) treats values that arrive there. 

If you can help me solve that riddle, I'd be willing to give this another shot. Otherwise, I already spent quite some time trying this out to no prevail :(

Reply:
> The data migration fails due to it trying to insert data into a non existent column

Ohhh, because the one-to-many field is virtual and is based on the existence of the row in directus_relations 🤔 Hmm, difficult problem! In order to get around that, we'd almost have to migrate all the data _without_ any of the relational fields, to make sure all the parent items exist, and then apply the relations, and then do another pass over the data to update the items with the relational values 🥴

Reply:
Or now that I think about it, one-to-many fields shouldn't have to be populated, as they're based on the other m2o value. Maybe if we simply skip over the one to many fields altogether things just work? 🤔

Reply:
Oh if that works that would be quite neat. So your suggestion would be to just skip o2m fields and just fill m2o ones? 

I can try this out. 

Probably only going to get around on Sunday to work on this. really busy work week :(

Reply:
> So your suggestion would be to just skip o2m fields and just fill m2o ones?

Yeah! That should do the trick 🙂 

> Probably only going to get around on Sunday to work on this. really busy work week :(

No worries! Same here 😭

Reply:
Hey @rijkvanzanten,
This doesn't really fix the issue when a collection with an m2o is inserted before the corresponding o2m collection, because then it cannot find the right items. 

I can still make this change, but imho it doesn't solve the problem I'm trying to solve with the sorting. 

Wdyt?